plugins/default/check_mongodb_wrapper.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+######################################################################
+#
+# Nagios wrapper plugin to retrieve mongo admin password from
+# same project where nagios is located and then perform the
+# check requested
+#
+######################################################################
+
+################################
+# GLOBALS
+################################
+# Nagios exit codes
+NAGIOS_OK=0
+NAGIOS_WARN=1
+NAGIOS_CRIT=2
+NAGIOS_UNKNOWN=3
+
+################################
+# FUNCTIONS
+################################
+usage() {
+ echo "Usage:"
+ echo " -w <warn>    Warning threshold (default 0)"
+ echo " -c <crit>    Critical threshold (default 0)"
+ echo " -A <action>  Action as defined"
+ echo " -S <name>     OpenShift selector name"
+ echo " -h           Show this message"
+ echo "Example: $0 -A connect -S mongodb-server -w 0 -c 0"
+}
+
+################################
+# ARGUMENT PROCESSING
+################################
+WARN=0
+CRIT=0
+ACTION=
+SELECTOR=mongodb-replica
+while getopts "h:w:c:A:S:" OPTION
+do
+ case $OPTION in
+  h) usage; exit $NAGIOS_UNKNOWN ;;
+  w) WARN=$OPTARG ;;
+  c) CRIT=$OPTARG ;;
+  A) ACTION=$OPTARG ;;
+  S) SELECTOR=$OPTARG ;;
+  ?) usage; exit $NAGIOS_UNKNOWN ;;
+ esac
+done
+
+# Ensure all expected variables are populated
+if [[ -z $ACTION ]] || [[ -z $SELECTOR ]] || [[ -z $WARN ]] || [[ -z $CRIT ]]; then
+ usage; exit $NAGIOS_UNKNOWN
+fi
+
+# Use a temp file to store list of pods with their mongodb connection info
+TEMP_FILE=`mktemp`
+oc get pod --selector=name=$SELECTOR --output json | jq '.items[] | select(.status.phase == "Running") | [.metadata.name, .status.podIP, .spec.containers[].ports[].containerPort, ( .spec.containers[].env[] | select(.name == "MONGODB_ADMIN_PASSWORD") | .value) ] | @tsv' -r > $TEMP_FILE
```

Comment:
I'd be in favor of splitting long lines, specially when there are important things towards the end of the line that could pass unnoticed.